pkg/deploy/api/validation/validation_test.go

```diff
@@ -79,6 +79,45 @@ func TestValidateDeploymentConfigOK(t *testing.T) {
 	}
 }
 
+func TestValidateDeploymentConfigICTMissingImage(t *testing.T) {
+	errs := ValidateDeploymentConfig(&api.DeploymentConfig{
+		ObjectMeta: kapi.ObjectMeta{Name: "foo", Namespace: "bar"},
+		Spec: api.DeploymentConfigSpec{
+			Replicas: 1,
+			Triggers: []api.DeploymentTriggerPolicy{test.OkImageChangeTrigger()},
+			Selector: test.OkSelector(),
+			Strategy: test.OkStrategy(),
+			Template: test.OkPodTemplateMissingImageWithOkImageChangeTrigger(),
+		},
+	})
+
+	if len(errs) > 0 {
+		t.Errorf("Unxpected non-empty error list: %+v", errs)
+	}
+}
+
+func TestValidateDeploymentConfigNoICTMissingImage(t *testing.T) {
+	errs := ValidateDeploymentConfig(&api.DeploymentConfig{
+		ObjectMeta: kapi.ObjectMeta{Name: "foo", Namespace: "bar"},
+		Spec: api.DeploymentConfigSpec{
+			Replicas: 1,
+			Triggers: []api.DeploymentTriggerPolicy{test.OkConfigChangeTrigger()},
+			Selector: test.OkSelector(),
+			Strategy: test.OkStrategy(),
+			Template: test.OkPodTemplateMissingImageWithOkImageChangeTrigger(),
+		},
+	})
+
+	if len(errs) == 0 {
+		t.Error("Should have gotten error list")
+	}
+
+	if got, exp := errs[0].Type, field.ErrorTypeRequired; got != exp {
+		t.Errorf("Instead of a required error got %v", got)
+	}
+
+}
```

Comment:
can't this  test be part of TestValidateDeploymentMissingFields  (and should check the actual field reported in the error, like those tests do)